### ISSUE TITLE
Qt package can't be current because of subpackage differences

### ISSUE THREAD

**Issue by jrmarino** [repology/repology-rules#451]
So Qt package contains both qt5 and qt6.  At first glance this is okay;
Qt5 would be considered a legacy branch.  So all qt5 versions at 5.15.2 would
be considered "orange current" and all qt6 versions at 6.0.0 would be considered
"green current".

Unfortunately, it's never going to work for the same reasons as #429.

There are many packages in qt5 that do not have qt6 counterparts.
For example, qtxmlpatterns is present only in qt5
see: http://download.qt.io/official_releases/qt/5.15/5.15.2/submodules/
see: http://download.qt.io/official_releases/qt/6.0/6.0.0/submodules/

So qtxmlpatterns can never be at a version higher than 5.15.2, so the entire
qt package would always be considered out of date for all repositories that
split qt into subpackages.

If Repology can't solve mismatching modules seen in PHP and Qt, then the
probably solution is to split Qt into separate package (qt5 and qt6).

As a second (related issue), this report was improperly rejected:

comment:
```
another date based version that should be ignored:
Gentoo dev-qt/qt-docs 5.15.2_p202011130614
```

reply:
```
There's no need to ignore non-latest versions
```

That's wrong.  
The top qt5 version is listed as 5.15.2_p202011130614, higher than 5.15.2, so
even if the qt5 legacy branch worked, it would show out of date.  So there is
a very real need to ignore date based versions (It was wrong to classify it
as the non-latest.  It's also just wrong in general, date based versions should
always be marked incorrect for accuracy).

So Qt is a real mess which started when all the qt packages were grouped as one
so something needs to happen to either make that work properly or as I said,
split out Qt into Qt5 and Qt6.

**Comment by AMDmi3** [comment]
Duplicate of #429

**Comment by jrmarino** [comment]
what about the second part?  where the date report was improperly rejected?

**Comment by AMDmi3** [comment]
`*p*` versions from Gentoo are ignored by default

**Comment by jrmarino** [comment]
Ok, I believe you.
However, that brings up a separate issue that has bitten me several times.
![image](https://user-images.githubusercontent.com/1125622/104505289-475c5700-55a9-11eb-8c63-463e99ed865b.png)
The "p" version shows as red, not gray, so it's misleading.  There is no way for an user to tell that version is ignored.  It's the reason when new repositories create the same mistake as those ignored by rule and all show, not the new one.   
I'd like to document this issue -- I'm not sure if you already have it elsewhere though.
